lib/src/components/power_sensor/power_sensor.dart

```diff
@@ -0,0 +1,84 @@
+import '../../components/sensor/sensor.dart';
+import '../../gen/common/v1/common.pb.dart';
+import '../../gen/component/powersensor/v1/powersensor.pb.dart';
+import '../../resource/base.dart';
+import '../../robot/client.dart';
+
+class Voltage {
+  double volts;
+  bool isAc;
+
+  Voltage(this.volts, this.isAc);
+
+  factory Voltage.fromProto(GetVoltageResponse pbResponse) {
+    return Voltage(pbResponse.volts, pbResponse.isAc);
+  }
+}
+
+class Current {
+  double amperes;
+  bool isAc;
+
+  Current(this.amperes, this.isAc);
+
+  factory Current.fromProto(GetCurrentResponse pbResponse) {
+    return Current(pbResponse.amperes, pbResponse.isAc);
+  }
+}
```

Comment:
Instead of creating these types which are identical to the response type (but with better names), you can just create `typedef`s that should do the same thing:

```dart
typedef Voltage = GetVoltageResponse;
typedef Current = GetCurrentResponse;
```

You will have to change a bit of the code, but this way you don't have to create brand new classes